src/Internal/Payment/PaymentContext.php

```diff
@@ -7,9 +7,92 @@
 
 namespace WCPay\Internal\Payment;
 
+use Automattic\WooCommerce\Blocks\Payments\PaymentMethodTypeInterface;
+use WCPay\Internal\Payment\PaymentMethod\PaymentMethodInterface;
+
 /**
  * A context object, which is shared between payment states.
  */
 class PaymentContext {
+	/**
+	 * Contains all the context's data.
+	 *
+	 * @var array
+	 */
+	private $data = [];
```

Comment:
Could we consider using well-defined variables in place of the $data array? Using well-defined variables offers several advantages, including type-safety, immutability (if necessary), and the ability to provide type-hinting. With $data, one can store anything, which may lead to unintended consequences. I believe it's important to be stringent in defining the PaymentContext, as it encompasses a critical piece of logic—data. Allowing arbitrary data storage in this context could pose potential challenges in the future.

Reply:
This comment seems to ignore the fact that `set()`, `get()` and `$data` are all private and can only be accessed through the strongly typed getters and setters.

> immutability (if necessary)

I don't think we require anything here to be immutable, do we? At some point, the context object will be shared between requests, and follow-up requests might provide new details. We want to use `set()` to make sure that those changes are logged, but we'll trust the latest input anyway.